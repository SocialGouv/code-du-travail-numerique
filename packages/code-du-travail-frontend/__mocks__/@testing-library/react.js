/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
