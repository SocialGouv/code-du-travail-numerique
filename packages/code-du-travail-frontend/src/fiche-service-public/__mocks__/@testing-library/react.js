import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
