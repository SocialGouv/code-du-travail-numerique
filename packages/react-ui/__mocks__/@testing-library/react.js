import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
