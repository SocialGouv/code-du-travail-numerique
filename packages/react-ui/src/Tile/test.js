import { render, screen } from "@testing-library/react";
import React from "react";

import { Time } from "../icons/index.js";
import { Tile } from "./index.js";

describe("<Tile />", () => {
  it("renders a tile with text inside", () => {
    const { container } = render(
      <Tile href="#" title="Hello !">
        There is text inside
      </Tile>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a tile with a custom badge", () => {
    const { container } = render(
      <Tile href="#" custom title="Hello !">
        There is text inside
      </Tile>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a tile with an icon", () => {
    const { container } = render(
      <Tile href="#" title="Hello !" icon={Time}>
        There is text inside
      </Tile>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a tile with a stripe", () => {
    const { container } = render(
      <Tile href="#" title="Hello !" striped>
        There is text inside
      </Tile>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a wide tile with a subtitle", () => {
    const { container } = render(
      <Tile wide subtitle="This is a theme" title="Hello !">
        There is text inside
      </Tile>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a tile with title with titleTagType", () => {
    render(
      <Tile href="#" title="Hello !" titleTagType="h2">
        There is text inside
      </Tile>
    );
    const title = screen.getByRole("heading", { level: 2 });
    expect(title.textContent).toEqual("Hello !");
  });
});
