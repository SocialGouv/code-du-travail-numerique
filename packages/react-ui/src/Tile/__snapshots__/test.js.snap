// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> renders a tile with a custom badge 1`] = `
.c1 {
  position: absolute;
  top: 0;
  right: 0;
  width: 5rem;
  height: 5rem;
  overflow: hidden;
  color: #fff;
  border-radius: 0 0.6rem 0 0;
}

.c1:before {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border-color: transparent #f66663 transparent transparent;
  border-style: solid;
  border-width: 0 5rem 5rem 0;
  content: "";
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  z-index: 1;
  width: 3rem;
  height: 3rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: 100%;
  margin: 0;
  padding: 3.2rem 2rem;
  color: #3e486e;
  font-weight: normal;
  font-size: 1.6rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: box-shadow 250ms linear, -webkit-transform 100ms linear;
  -webkit-transition: box-shadow 250ms linear, transform 100ms linear;
  transition: box-shadow 250ms linear, transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #3e486e;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.4);
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c4 {
  padding-right: 1rem;
}

.c5 {
  margin: 0;
  font-size: 1.8rem;
  font-weight: 600;
}

.c6 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 1rem;
}

@media (max-width:600px) {
  .c0 {
    padding: 2rem 1.6rem;
    font-size: 1.4rem;
  }
}

<div>
  <a
    class="c0"
    href="#"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <svg
          aria-hidden="true"
          fill="none"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.938 16.5a3.548 3.548 0 0 1 3.562-3.563 3.548 3.548 0 0 1 3.563 3.563 3.548 3.548 0 0 1-3.563 3.563 3.548 3.548 0 0 1-3.563-3.563zm2.062 0c0 .84.66 1.5 1.5 1.5s1.5-.66 1.5-1.5-.66-1.5-1.5-1.5-1.5.66-1.5 1.5z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M25 14.838v3.404l-1.663.87-.474 1.109.554 1.82-2.375 2.376-1.9-.555-1.109.476-.87 1.662h-3.405l-.87-1.663-1.109-.474-1.82.554-2.376-2.375.554-1.9-.475-1.109L6 18.163v-3.405l1.662-.87.475-1.109-.554-1.82 2.375-2.376 1.9.554 1.109-.475.87-1.662h3.405l.87 1.662 1.109.475 1.82-.554 2.376 2.375-.555 1.9.476 1.109 1.662.87zm-4.1 5.112.825-2.025L23 17.25v-1.5l-1.2-.6-.825-2.025.375-1.35-1.05-1.05-1.35.375-2.025-.825L16.25 9h-1.5l-.6 1.2-2.025.825-1.35-.375-1.05 1.05.375 1.35-.825 2.025L8 15.75v1.5l1.2.6.825 2.025-.375 1.35 1.05 1.05 1.35-.375 2.025.825L14.75 24h1.5l.6-1.2 2.025-.825 1.35.375 1.05-1.05-.375-1.35z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        
        <p
          class="c5"
        >
          Hello !
        </p>
      </div>
    </div>
    <div
      class="c6"
    >
      There is text inside
    </div>
  </a>
</div>
`;

exports[`<Tile /> renders a tile with a stripe 1`] = `
.c2 {
  position: absolute;
  background-color: #7994d4;
  border-radius: 0;
  top: 0;
  left: 50%;
  width: 5rem;
  height: 0.4rem;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: 100%;
  margin: 0;
  padding: 3.2rem 2rem;
  color: #3e486e;
  font-weight: normal;
  font-size: 1.6rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: box-shadow 250ms linear, -webkit-transform 100ms linear;
  -webkit-transition: box-shadow 250ms linear, transform 100ms linear;
  transition: box-shadow 250ms linear, transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #3e486e;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.4);
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  padding-right: 1rem;
}

.c4 {
  margin: 0;
  font-size: 1.8rem;
  font-weight: 600;
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 1rem;
}

@media (max-width:600px) {
  .c0 {
    padding: 2rem 1.6rem;
    font-size: 1.4rem;
  }
}

<div>
  <a
    class="c0"
    href="#"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      />
      <div
        class="c3"
      >
        
        <p
          class="c4"
        >
          Hello !
        </p>
      </div>
    </div>
    <div
      class="c5"
    >
      There is text inside
    </div>
  </a>
</div>
`;

exports[`<Tile /> renders a tile with an icon 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: 100%;
  margin: 0;
  padding: 3.2rem 2rem;
  color: #3e486e;
  font-weight: normal;
  font-size: 1.6rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: box-shadow 250ms linear, -webkit-transform 100ms linear;
  -webkit-transition: box-shadow 250ms linear, transform 100ms linear;
  transition: box-shadow 250ms linear, transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #3e486e;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.4);
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c2 {
  display: block;
  width: 7.2rem;
  height: 7.2rem;
  margin: 0 auto 1.6rem;
  padding: 1.4rem;
  background-color: #f2f5fa;
  border-radius: 50%;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  padding-right: 1rem;
}

.c4 {
  margin: 0;
  font-size: 1.8rem;
  font-weight: 600;
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 1rem;
}

@media (max-width:600px) {
  .c0 {
    padding: 2rem 1.6rem;
    font-size: 1.4rem;
  }
}

<div>
  <a
    class="c0"
    href="#"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <svg
          aria-hidden="true"
          fill="none"
          viewBox="0 0 52 52"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M44.385 7.615C39.474 2.705 32.945 0 26 0 19.055 0 12.526 2.704 7.615 7.615 2.705 12.526 0 19.055 0 26c0 6.945 2.705 13.474 7.615 18.385C12.526 49.295 19.055 52 26 52c6.945 0 13.474-2.704 18.385-7.615C49.295 39.474 52 32.945 52 26c0-6.945-2.705-13.474-7.615-18.385zM26 49.97C12.784 49.969 2.031 39.216 2.031 26S12.784 2.031 26 2.031c13.216 0 23.969 10.753 23.969 23.969 0 13.216-10.753 23.969-23.969 23.969z"
            fill="#4E6896"
          />
          <path
            d="M34.71 5.86a21.916 21.916 0 0 1 12.832 15.971 1.016 1.016 0 0 1-1.995.384 19.886 19.886 0 0 0-11.644-14.49 1.016 1.016 0 1 1 .807-1.865zm-4.021-1.295.026.006a1.016 1.016 0 1 1-.442 1.982l-.012-.002a1.016 1.016 0 1 1 .428-1.986zM6.453 29.785a19.853 19.853 0 0 0 11.873 14.589 1.015 1.015 0 1 1-.783 1.874 22.062 22.062 0 0 1-8.521-6.355 21.938 21.938 0 0 1-4.564-9.724 1.016 1.016 0 0 1 1.995-.384zm15.225 15.651.063.014a1.016 1.016 0 0 1-.432 1.985l-.071-.016a1.016 1.016 0 1 1 .44-1.983zM45.45 30.259a1.016 1.016 0 0 1 1.984.432 21.849 21.849 0 0 1-6.104 11.002 21.836 21.836 0 0 1-11.161 5.849 1.016 1.016 0 0 1-.384-1.995A19.854 19.854 0 0 0 45.45 30.26zM21.831 4.458a1.015 1.015 0 1 1 .384 1.995A19.854 19.854 0 0 0 6.55 21.741a1.016 1.016 0 0 1-1.985-.432 21.848 21.848 0 0 1 6.105-11.002 21.836 21.836 0 0 1 11.161-5.849z"
            fill="#FF7067"
          />
          <path
            clip-rule="evenodd"
            d="M27.016 22.7v-8.65a1.016 1.016 0 0 0-2.031 0v8.65a3.469 3.469 0 0 0-2.286 2.285h-3.774a1.016 1.016 0 0 0 0 2.031h3.774A3.46 3.46 0 0 0 26 29.453 3.457 3.457 0 0 0 29.453 26a3.46 3.46 0 0 0-2.437-3.3zm-1.037 1.878a1.513 1.513 0 0 0 .042 0A1.423 1.423 0 0 1 26 27.422 1.423 1.423 0 0 1 24.578 26v-.02a1.423 1.423 0 0 1 1.401-1.402z"
            fill="#FF7067"
            fill-rule="evenodd"
          />
          <path
            d="M25 5.078a1.016 1.016 0 0 1 2.031 0v2.276a1.016 1.016 0 1 1-2.031 0V5.078zm-.031 38.844a1.016 1.016 0 0 1 2.031 0v3a1.016 1.016 0 0 1-2.031 0v-3zM44.18 25h2.742a1.016 1.016 0 0 1 0 2.031H44.18a1.016 1.016 0 0 1 0-2.031zm-39.102-.031h2.945a1.016 1.016 0 0 1 0 2.031H5.078a1.016 1.016 0 0 1 0-2.031z"
            fill="#FF7067"
          />
        </svg>
      </div>
      <div
        class="c3"
      >
        
        <p
          class="c4"
        >
          Hello !
        </p>
      </div>
    </div>
    <div
      class="c5"
    >
      There is text inside
    </div>
  </a>
</div>
`;

exports[`<Tile /> renders a tile with text inside 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: 100%;
  margin: 0;
  padding: 3.2rem 2rem;
  color: #3e486e;
  font-weight: normal;
  font-size: 1.6rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: box-shadow 250ms linear, -webkit-transform 100ms linear;
  -webkit-transition: box-shadow 250ms linear, transform 100ms linear;
  transition: box-shadow 250ms linear, transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #3e486e;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.4);
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c2 {
  padding-right: 1rem;
}

.c3 {
  margin: 0;
  font-size: 1.8rem;
  font-weight: 600;
}

.c4 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 1rem;
}

@media (max-width:600px) {
  .c0 {
    padding: 2rem 1.6rem;
    font-size: 1.4rem;
  }
}

<div>
  <a
    class="c0"
    href="#"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        
        <p
          class="c3"
        >
          Hello !
        </p>
      </div>
    </div>
    <div
      class="c4"
    >
      There is text inside
    </div>
  </a>
</div>
`;

exports[`<Tile /> renders a wide tile as a button with a subtitle 1`] = `
.c3 {
  display: block;
  margin: 0 0 1rem 0;
  color: #4d73b8;
  font-weight: 600;
  font-size: 1.2rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.625;
  text-transform: uppercase;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  margin: 0;
  padding: 2rem 2rem;
  color: #3e486e;
  font-weight: normal;
  font-size: 1.6rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: box-shadow 250ms linear, -webkit-transform 100ms linear;
  -webkit-transition: box-shadow 250ms linear, transform 100ms linear;
  transition: box-shadow 250ms linear, transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #3e486e;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.4);
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c2 {
  padding-right: 1rem;
}

.c4 {
  margin: 0;
  font-size: 1.8rem;
  font-weight: 600;
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 1rem;
}

@media (max-width:600px) {
  .c0 {
    padding: 1.6rem 1.6rem;
    font-size: 1.4rem;
  }
}

<div>
  <button
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3 "
        >
          This is a theme
        </p>
        <p
          class="c4"
        >
          Hello !
        </p>
      </div>
    </div>
    <div
      class="c5"
    >
      There is text inside
    </div>
  </button>
</div>
`;
